compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -207,18 +208,28 @@ public void save() {
             writeBuildFile(buildFilePath);
             writeDependencies();
         } else {
-            // build file exists
-            BuildJson buildJson = readBuildJson(buildFilePath);
-            // need to update Dependencies toml
-            writeDependencies();
-            // check whether last updated time is expired
-            if (shouldUpdate) {
-                // update build json file
-                writeBuildFile(buildFilePath);
-            } else {
-                // only update build time
-                buildJson.setLastBuildTime(System.currentTimeMillis());
-                ProjectUtils.writeBuildFile(buildFilePath, buildJson);
+            try {
+                // build file exists
+                BuildJson buildJson = readBuildJson(buildFilePath);
+                // need to update Dependencies toml
+                writeDependencies();
+                if (buildJson != null) {
+                    // check whether last updated time is expired
+                    if (shouldUpdate) {
+                        // update build json file
+                        writeBuildFile(buildFilePath);
+                    } else {
+                        // only update build time
+                        buildJson.setLastBuildTime(System.currentTimeMillis());
+                        ProjectUtils.writeBuildFile(buildFilePath, buildJson);
+                    }
+                } else {
+                    writeBuildFile(buildFilePath);
+                }
+            } catch (JsonSyntaxException e) {
```

Comment:
Shouldn't we ignore or report a warning instead of an exception here?

Reply:
Since this is not important to the user, I think we can ignore and rewrite the file. Also, we should be able to add a no-location diagnostics to report a warning which we don't have atm.

Reply:
+1 to ignore since this file should be transparent to user

Reply:
With 864788ea19e77a6292976f2b999a8c933034371d we ignore the JSONSyntaxException and simply rewrite the build file